resolver/resolver.cc

```diff
@@ -312,18 +312,29 @@ class ResolveConstantsWalk {
 
         bool couldDefineChildNamespace(const core::GlobalState &gs, const std::vector<core::NameRef> &prefix,
                                        const std::vector<ast::ConstantLit *> &suffix) const {
+            ENFORCE(!prefix.empty());
+
+            auto start = prefix.begin();
+            if (*start == core::Names::Constants::Test()) {
+                ++start;
+            }
```

Comment:
That's correct. The `Test::` prefix is synthetic, and present implicitly for every package.

Reply:
There's currently no error added from processing a `__package.rb` that defines a package in `Test::`, but you do get an error once you start trying to define classes in that package. I think I'll make a separate PR to add an error for packages defined in `Test::`.

Reply:
Currently, defining a package in the `Test::` namespace causes `ENFORCE` failures in the resolver. It might be a little more tricky to add an error than I thought.